homeassistant/components/shelly/switch.py

```diff
@@ -71,6 +80,23 @@ class BlockSwitchDescription(BlockEntityDescription, SwitchEntityDescription):
     entity_category=EntityCategory.CONFIG,
 )
 
+RPC_SWITCHES = {
+    "switch": RpcSwitchDescription(
+        key="switch",
+        sub_key="output",
+        entity_category=EntityCategory.CONFIG,
+        api_call_on="Switch.Set",
+        api_call_off="Switch.Set",
+    ),
+    "script": RpcSwitchDescription(
+        key="script",
+        sub_key="running",
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
Commenting so it is not accidently merged, as we discussed without changes `entity_category` doesn't work in this implementation.

Reply:
Changed PR to introduce script switch only. Rework will be done in a separate PR

Reply:
Entity category is working... why shouldn't we use the `diagnostic` category?

![obraz](https://github.com/user-attachments/assets/47393c0d-4cbb-40e5-a8a2-1fef43adbe33)